tools/github-team-user-store/GitHubTeamUserStore/GitHubTeamUserStore/TeamUserGenerator.cs

```diff
@@ -0,0 +1,133 @@
+using Octokit;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Text.Json;
+using System.Threading.Tasks;
+using GitHubTeamUserStore.Constants;
+
+namespace GitHubTeamUserStore
+{
+    public class TeamUserGenerator
+    {
+        // Team/User dictionary where the team name is the key and the list users is the value. It's worth noting
+        // that this is the list of Logins (Octokit.User.Login) which are what's used in CODEOWNERS, @mentions, etc.
+        private static Dictionary<string, List<string>> _teamUserDict = new Dictionary<string, List<string>>();
+
+        /// <summary>
+        /// Generate the team/user lists for each and every team under azure-sdk-write. Every team and user in a CODEOWNERS
+        /// file must have azure-sdk-write permissions in order to be in there which means every team/user will be under
+        /// of azure-sdk-write. This is done to limit the number of calls made because if team/user data for every team
+        /// under Azure was pulled there would be roughly 1933 teams. Getting the team/user data for that many teams would
+        /// end up taking 2500-3000 GitHub API calls whereas getting the team/user data for azure-sdk-write and its child
+        /// teams is less than 1/10th of that. The team/user data is serialized into json and stored in azure blob storage.
+        /// </summary>
+        /// <param name="gitHubEventClient">Authenticated GitHubEventClient</param>
+        /// <returns></returns>
+        public static async Task GenerateTeamUserList(GitHubEventClient gitHubEventClient)
+        {
+            Team azureSdkWrite = await gitHubEventClient.GetTeamById(ProductAndTeamConstants.AzureSdkWriteTeamId);
+            await GetUsersForTeam(gitHubEventClient, azureSdkWrite);
+            // Serializing the Dictionary<string, List<string>> directly won't work with the JsonSerializer but
+            // a List<KeyValuePair<string, List<string>>> will and it's easy enough to convert to/from.
+            var list = _teamUserDict.ToList();
+            string jsonString = JsonSerializer.Serialize(list);
+            await gitHubEventClient.UploadToBlobStorage(jsonString);
+        }
+
+        /// <summary>
+        /// Call GitHub to get users for the team and add a dictionary entry for the team/users. Note: GitHub returns 
+        /// a distinct list of all of the users, including users from any/all child teams. After that, get the list
+        /// of child teams and recursively call GetUsersForTeam for each one of those to create their team/user entries.
+        /// </summary>
+        /// <param name="gitHubEventClient">Authenticated GitHubEventClient</param>
+        /// <param name="team">Octokit.Team to get users for.</param>
+        public static async Task GetUsersForTeam(GitHubEventClient gitHubEventClient, Team team)
```

Comment:
Given this method starts with `Get` I would expect for it to return some value, but it returns nothing.

Reply:
Renamed to CreateTeamUserEntry